requirements.pinned.txt

```diff
@@ -1,177 +1,185 @@
-alembic==1.7.7
-amqp==5.1.1
-arrow==1.2.3
-async-timeout==4.0.2
-attrs==22.1.0
-Babel==2.11.0
-backcall==0.2.0
+alembic==1.10.4
+amqp==5.2.0
+appdirs==1.4.4
+arrow==1.3.0
+asttokens==2.4.1
+async-timeout==4.0.3
+attrs==23.2.0
+Babel==2.14.0
 base32-lib==1.0.2
 billiard==3.6.4.0
-bleach==4.1.0
-blinker==1.5
-cached-property==1.5.2
-cachelib==0.6.0
+bleach==6.1.0
+blinker==1.7.0
+cachelib==0.9.0
 cds-dojson==0.9.0
-celery==5.1.2
-certifi==2022.9.24
-cffi==1.15.1
-charset-normalizer==2.0.12
-click==7.1.2
-click-default-group==1.2.2
+celery==5.2.7
+certifi==2023.11.17
+cffi==1.16.0
+charset-normalizer==3.3.2
+click==8.1.7
+click-default-group==1.2.4
 click-didyoumean==0.3.0
 click-plugins==1.1.1
-click-repl==0.2.0
+click-repl==0.3.0
 counter-robots==2018.6
-cryptography==38.0.3
-dataclasses==0.8
+cryptography==41.0.7
 decorator==5.1.1
-Deprecated==1.2.13
-dnspython==2.2.1
-dojson==1.4.0
-elasticsearch==7.13.4
-elasticsearch-dsl==7.4.0
-email-validator==1.3.0
-Flask==2.0.3
-Flask-Admin==1.6.0
+dictdiffer==0.9.0
+dnspython==2.4.2
+dojson==1.5.0
+email-validator==2.1.0.post1
+exceptiongroup==1.2.0
+executing==2.0.1
+Flask==2.2.5
+Flask-Admin==1.6.1
 Flask-Alembic==2.0.1
-Flask-BabelEx==0.9.4
+flask-babel==4.0.0
 Flask-Breadcrumbs==0.5.1
-Flask-Caching==1.10.1
-Flask-CeleryExt==0.4.3
+Flask-Caching==2.1.0
+Flask-CeleryExt==0.5.0
 Flask-Collect-Invenio==1.4.0
-Flask-Cors==3.0.10
-Flask-DebugToolbar==0.13.1
+Flask-Cors==4.0.0
+Flask-DebugToolbar==0.14.1
 Flask-KVSession-Invenio==0.6.3
 Flask-Limiter==1.1.0
-Flask-Login==0.4.1
+Flask-Login==0.6.3
 Flask-Mail==0.9.1
-Flask-Menu==0.7.2
+Flask-Menu==0.7.3
 Flask-OAuthlib==0.9.6
 Flask-Principal==0.4.0
-Flask-Security==3.0.0
-flask-shell-ipython==0.4.1
-Flask-SQLAlchemy==2.4.4
+Flask-Security-Invenio==3.3.3
+flask-shell-ipython==0.5.1
+Flask-SQLAlchemy==2.5.1
 flask-talisman==0.8.1
 flask-webpackext==1.0.2
-Flask-WTF==1.0.1
-fs==0.5.4
-ftfy==6.0.3
-future==0.18.2
+Flask-WTF==1.2.1
+fs==2.4.16
+ftfy==6.1.3
+future==0.18.3
 fuzzywuzzy==0.18.0
-idna==3.4
-importlib-metadata==4.8.3
-importlib-resources==5.4.0
+github3.py==4.0.1
+greenlet==3.0.3
```

Comment:
are we sure about `greenlet`?

Reply:
thats a good question, it gets installed with some other dependency, but pip does not want to disclose which dependency installs it:
<img width="1002" alt="Screenshot 2024-01-19 at 17 21 41" src="https://github.com/CERNDocumentServer/cds-ils/assets/38131488/dec0c6c1-776d-4f96-9e68-7ca4a83a59d8">